app/controllers/event_instances_controller.rb

```diff
@@ -3,14 +3,18 @@ class EventInstancesController < ApplicationController
   before_filter :cors_preflight_check, except: [:index]
 
   def update
-    event_instance = EventInstance.find_or_create_by(uid: params[:id])
+    unless EventInstance.find_by(uid: params[:id])
+      EventInstance.create(uid: params[:id], event_id: params[:event_id])
+    end
+    event_instance = EventInstance.find_by(uid: params[:id])
+    # another option could be to just call a different method on the instance.
 
     event_instance_params = check_and_transform_params(event_instance)
     hangout_url_changed = event_instance.hangout_url != event_instance_params[:hangout_url]
     yt_video_id_changed = event_instance.yt_video_id != event_instance_params[:yt_video_id]
     slack_notify = params[:notify] == 'true'
 
-    if event_instance.try!(:update, event_instance_params)
+    if event_instance.try!(:update, event_instance_params) #why would it not update.  is it only if things changed?
```

Comment:
I think this is caution because the data is coming in from a google hangout plugin and we're trying to avoid a failure on the off chance that event_instance would be nil - but that said this is really gnarly code -- there's another PR coming through that cleans up a little here so I'd be cautious about working on this  - I might suggest that it would be more effective to review the existing open PRs and get your head around what other team members are doing than trying to make code changes here.

That said, don't let me stop you playing with experimental PRs in any part of the codebase, just that in some areas where existing work is going on the changes of challenging merge conflicts is much higher than others ...